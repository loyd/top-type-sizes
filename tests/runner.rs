use top_type_sizes::*;

macro_rules! test_sample {
    ($name:ident) => {
        #[test]
        fn $name() {
            static SAMPLE: &'static str =
                include_str!(concat!("samples/", stringify!($name), ".txt"));

            let refined_sample = reader::read(SAMPLE.as_bytes()).unwrap();
            let types = parser::parse(&refined_sample).unwrap();
            insta::assert_debug_snapshot!(format!("{}_types", stringify!($name)), types);

            // TODO: check the transformer.

            let output = formatter::format(types, &Default::default());
            println!("{output}");
            // TODO: make snapshots of output.
        }
    };
}

test_sample!(command);
test_sample!(control_flow);
test_sample!(alignment_enum);
test_sample!(tokio_udp);
test_sample!(several_types);
test_sample!(timex);
test_sample!(async_fn);
test_sample!(compiler_messages);

// TODO: add samples from rustc tests.
