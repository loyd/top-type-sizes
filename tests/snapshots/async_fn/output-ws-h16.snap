---
source: tests/runner.rs
description: top-type-sizes -ws -h16
---
16386 [async fn body@$DIR/async.rs:8:36: 11:2] align=1
  16385 variant Suspend0
       8192 arg (upvar) align=1 offset=0
       8192 arg
   8192 variant Unresumed, Returned, Panicked
       8192 arg (upvar) align=1 offset=0
